apps/mobile/src/components/settings/settings-list-item.tsx

```diff
@@ -15,7 +15,7 @@ interface ItemBaseProps {
   type?: ItemType;
   title: string;
   caption?: string;
-  icon?: ReactNode;
+  icon?: ReactElement;
```

Comment:
I guess this is enforcing it must be a component?

Reply:
Exactly–a narrower type. ReactNode is a broad superset that includes this, + `string | number | boolean | null | undefined` and probably more, none of which is applicable here. We could technically narrow this even further limiting which elements exactly can be passed (i.e. just our icons), but too much strictness is often a footgun.